### PR TITLE
Yet another ibm floating ip fix

### PR DESCRIPTION
**Pull request by a-dubs**
do not squash - keep 3 separate commits

### CODE REVIEW

**Review by TheRealFalcon — COMMENTED** [review]
One major comment inline

File: pycloudlib/ibm/instance.py [hunk 1]
```diff
@@ -913,7 +917,7 @@ def _wait_for_instance_start(self, **kwargs):
             side_effect_fn=self._check_instance_failed_status,
         )
 
-    def wait_for_delete(self, sleep_seconds=30, raise_on_fail=False):
+    def wait_for_delete(self, sleep_seconds=300, raise_on_fail=False):
```
[the remark this answers is not in the document]
Comment (a-dubs) [reply pycloudlib/ibm/instance.py:920]: We've had many instances recently take a couple minutes to finish deleting, and we can't clean up ssh keys until they delete. I'd rather just wait for the instance to finish delete than exit early and leave orphaned ssh keys

File: pycloudlib/ibm/instance.py [hunk 2]
```diff
@@ -597,10 +597,14 @@ def _attach_floating_ip(self, floating_ip: dict) -> bool:
             instance_id=self._instance["id"],
             network_interface_id=nic_id,
         ).get_result()
-        # sleep 1s to let cloud update then check to make sure that floating IP successfully attached
-        time.sleep(1)
-        self._floating_ip = self._discover_floating_ip(self._client, self._instance)
-        return self._floating_ip is not None
+        # sleep 10s to let cloud update and resolve any race conditions with other instances
+        self._log.debug("Sleeping 10s before retrieving floating ip status.")
+        time.sleep(10)
```
[the remark this answers is not in the document]
Comment (a-dubs) [reply pycloudlib/ibm/instance.py:602]: The problem is, it's not a matter of being ready. The API will simultaneously say two different instances have the floating IP attached when they query it, and so they think they're good to go and just proceed but only one of them will actually have the floating IP attached and then the other one will hang and eventually time out.

[the remark this answers is not in the document]
Comment (TheRealFalcon) [reply pycloudlib/ibm/instance.py:602]: Yikes...that's annoying. Is this a problem that could be solved by querying if the floating IP is already allocated, or is it just a race where IBM just doesn't know the current state? 

Are you seeing this in cloud-init integration tests, or somewhere else? If cloud-init, race conditions shouldn't really be an issue if tests are running serially.

[the remark this answers is not in the document]
Comment (a-dubs) [reply pycloudlib/ibm/instance.py:602]: It's something we are seeing in CPC Jenkins jobs where a handful of tests are kicked off simultaneously/in parallel. But yeah as you said, it's a race condition where IBM itself doesn't know which instance has the floating IP until after a few seconds until it settles it and resolves which instance actually has the floating IP.

[the remark this answers is not in the document]
Comment (TheRealFalcon) [reply pycloudlib/ibm/instance.py:602]: Gotcha. Well with cloud-init having a couple hundred integration tests (though not all require a launch), adding 10 seconds per launch will add hours to our IBM jobs. 

Can you add the context you provided here to the comment message? I think it's probably worth an issue to consider alternative solutions...could be something like an allocated floating ip list somewhere, or some other way of ensuring no collisions.

**Review by TheRealFalcon — APPROVED** [review]
LGTM, just see my other comment.

File: pycloudlib/ibm/instance.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (a-dubs) [reply pycloudlib/ibm/instance.py:602]: That is a good point. I don't want to incur a heavy penalty like that for you guys so I will reduce the wait time down to 3s (1s was just far too short. 3s should be good enough >90% of the time a race condition occurs) and also leave better context inline and in the commit message.

**Pull request merged**